src/Query/TimeoutExecutor.php

```diff
@@ -12,7 +12,12 @@ final class TimeoutExecutor implements ExecutorInterface
     private $loop;
     private $timeout;
 
-    public function __construct(ExecutorInterface $executor, $timeout, LoopInterface $loop = null)
+    /**
+     * @param ?ExecutorInterface $executor
+     * @param float|int $timeout
+     * @param ?LoopInterface $loop
+     */
+    public function __construct(?ExecutorInterface $executor, $timeout, ?LoopInterface $loop = null)
```

Comment:
Looks like the `$executor` was accidentally marked as nullable here. PHP also allows passing `int` to a `float` type, so should probably be simplified to just `float` here (plus native types in #225).

```suggestion
    /**
     * @param ExecutorInterface $executor
     * @param float $timeout
     * @param ?LoopInterface $loop
     */
    public function __construct(ExecutorInterface $executor, $timeout, ?LoopInterface $loop = null)
```

Reply:
Whoops that wasn't my intention. Still prefer keep these docblocks as small as possible as mention in https://github.com/reactphp/cache/pull/60#discussion_r1606090506 and better shown in https://github.com/reactphp/dns/pull/224#discussion_r1617497575. In this case just had an update for all of them.